api/response/schema_query.py

```diff
@@ -154,20 +154,25 @@ def search_response_by_room(self, info, **kwargs):
 
     def get_all_responses(self, info):
         response = []
-        rooms = RoomModel.query.all()
-        for room in rooms:
-            room_name = room.name
-            room_response = Response.get_query(info).filter_by(room_id=room.id)
-            response_count = room_response.count()
-            all_responses = Query.get_room_response(
-                self, room_response, room.id)
-            responses = RoomResponse(
-                room_id=room.id,
-                room_name=room_name,
-                total_responses=response_count,
-                response=all_responses)
-            response.append(responses)
-        return response
+        rooms = RoomModel.query.filter(RoomModel.state == "active").all()
+        if rooms:
+            for room in rooms:
+                room_name = room.name
+                room_response = Response.get_query(info).filter_by(
+                    room_id=room.id)
+                response_count = room_response.count()
+                all_responses = Query.get_room_response(
+                    self, room_response, room.id)
+                responses = RoomResponse(
+                    room_id=room.id,
+                    room_name=room_name,
+                    total_responses=response_count,
+                    response=all_responses)
+                response.append(responses)
+            return response
+        else:
+            raise GraphQLError(
+                'No response for this room, enter a valid room name')
```

Comment:
@d-kahara I believe this error message is what the user gets whenever there are no active rooms and the user wants to get `allRoomResponses`. I think the message does not describe the situation well enough as it implies the user entered a `room name` in the query argument. Kindly look into this

Reply:
This message is shown when the user provides a non-active `room name` as an argument in the query. See updated screenshots